docs/McpClient-Feedback.md

```diff
@@ -0,0 +1,47 @@
+ModelContextProtocol.Protocol.Implementation name is too generic.  Lots of things in this namespace named poorly.  That might be OK if users don't have to deal with them, but it looks like they do.
+
+
+ModelClient.CompleteAsync - naming seems off.  Maybe "GetCompletion" 
+
+
+
+static ModelClient.CreateSamplingHandler - why is this factory method on ModelClient???  Maybe should be an extension method.
+
+
+
+ModelClient - Enumerate\* vs List\* - why have these redundant methods that return the same thing just one wrapping.  Can we just return IAE and have folks materialize that?
```

Comment:
The List one is the vastly most common need, to actually get back an IList. IIRC, @eiriktsarpalis also thought it would be desirable to have the EnumerateXxAsync ones. I wouldn't want to drop the ListXxAsync ones, as that would be penalizing the 90% case.